### DISCUSSION TITLE
Inspecting rules locally when sync server is configured

### DISCUSSION THREAD

**Discussion by lid** [google/santa#1258] [Q&A] [answered]
I'd like to debug rules and verify that a client has retrieved the rules we expect from a sync server. What's the best way to inspect which rules santa has cached locally? 

What I've tried:
- `santactl status` shows that there are rules in the database, but I'd like to see what those are. 
- `sudo santactl rule --export` generates an error: `(SyncBaseURL/StaticRules is set, rules are managed centrally.)`

**Comment by mlw** [comment, marked as answer]
To debug, you can read the rules database. But note the file is locked if Santa is running. If that's the case you can simply copy it elsewhere first.

```
$ sudo cp /var/db/santa/rules.db /tmp/rules.db
$ sudo sqlite3 /tmp/rules.db
```

**Comment by lid** [comment]
Awesome, thanks! As a 1-liner to dump the rules and ignore database lock: 

```
echo -e '.mode column\n.headers on\nselect * from rules;\n' | sudo sqlite3 'file:/var/db/santa/rules.db?immutable=1'
```
